### ISSUE TITLE
Intra-doc links are unresolved on `pub use _ as _` items

### ISSUE THREAD

**Issue by Veetaha** [rust-lang/rust#76073]
I tried this code:

```rust
/// [`std::fs`]
pub use foo as foo2;

pub fn foo(){}

fn main() {}
```

I expected to see this happen:
If I run `cargo +nightly doc` I get no warnings/errors.

Instead, this happened:
```bash
$ cargo +nightly doc
 Documenting foo v0.1.0 (/home/veetaha/junk/rust-sandbox)
warning: unresolved link to `std::fs`
 --> src/main.rs:1:6
  |
1 | /// [`std::fs`]
  |      ^^^^^^^^^ unresolved link
  |
  = note: `#[warn(broken_intra_doc_links)]` on by default
  = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: 1 warning emitted
```

Note that if I remove `pub` from the `use` statement, no warnings are emitted...

### Meta
`rustc --version --verbose`:
```
rustc 1.47.0-nightly (2d8a3b918 2020-08-26)
binary: rustc
commit-hash: 2d8a3b9181f41d3af9b9f016c5d73b2553e344bf
commit-date: 2020-08-26
host: x86_64-unknown-linux-gnu
release: 1.47.0-nightly
LLVM version: 11.0
```
`rustdoc --version --verbose`:
```
rustdoc 1.47.0-nightly (2d8a3b918 2020-08-26)
binary: rustdoc
commit-hash: 2d8a3b9181f41d3af9b9f016c5d73b2553e344bf
commit-date: 2020-08-26
host: x86_64-unknown-linux-gnu
release: 1.47.0-nightly
LLVM version: 11.0
```

**Comment by jyn514** [comment]
```
$ RUSTDOC_LOG=rustdoc::passes::collect=debug rustdoc +stage1 pub-use.rs 
2020-08-29T20:03:27.412092Z DEBUG rustdoc::passes::collect_intra_doc_links: attempting to resolve item without parent module: std::fs
warning: unresolved link to `std::fs`
 --> pub-use.rs:1:6
  |
1 | /// [`std::fs`]
  |      ^^^^^^^^^ unresolved link
```
This is a bug in rustdoc, it should resolve relative to the current module.

**Comment by jyn514** [comment]
This will break #75756 until fixed.

**Comment by jyn514** [comment]
For some reason `tcx.parent()` returns `None` on `pub use` items 🤔

**Comment by jyn514** [comment]
Hmm, even after that's fixed it doesn't display the documentation anywhere ... @Veetaha did you expect it to show up above the re-export?
![image](https://user-images.githubusercontent.com/23638587/91645678-0d811e80-ea15-11ea-9930-ce7665d3a2b1.png)

Or on the index page for the module?
![image](https://user-images.githubusercontent.com/23638587/91645684-1eca2b00-ea15-11ea-8725-e20593b5edc1.png)

**Comment by Veetaha** [comment]
Well, I would expect it to appear at least somewhere :D
Might it be an oversight in rust or rustdoc design that it is allowed to write doc comments which on items where these doc comments are not rendered?
At the very least (even if unrendered) I would expect validation of the intra-doc link happening...

**Comment by jyn514** [comment]
Right, https://github.com/rust-lang/rust/pull/76082 fixes the incorrect error. If you don't have a preference I might put it on the `pub use`, although I wonder if maybe rustdoc should just give an error instead that you tried to put docs on a re-export.

**Comment by Veetaha** [comment]
Reexports are part of the crate's public API. I think it is reasonable to apply docs to them and have them rendered. We may collect more opinions on that before further decisions

**Comment by jyn514** [comment]
Going to close this since the intra-doc issue is fixed - if you think rustdoc should show documentation on pub re-exports of items within your crate, that's a separate issue (but feel free to open a new issue for it!)
